### PR TITLE
fix: month filter chart

### PR DESCRIPTION
**Pull request by rephus**
Closes: #4983

### Description:

TL;DR the chart query with dashboard filters was missing the timezone fix. 

Before:
Screenshot from 2023-04-11 15-09-02

After:
Screenshot from 2023-04-11 15-15-36

+ Bonus: I added and fixed some tests 

image

### CODE REVIEW

File: packages/backend/src/database/seeds/development/02_saved_queries.ts [hunk 1]
```diff
@@ -293,4 +294,57 @@ export async function seed(knex: Knex): Promise<void> {
             updatedByUser,
         },
     );
+
+    await savedChartModel.create(
```
Comment (rephus) [inline packages/backend/src/database/seeds/development/02_saved_queries.ts:298]: Added a new chart with `chart filters` to test a few different cases on e2e tests

File: packages/e2e/cypress/e2e/dates.cy.ts [hunk 2]
```diff
@@ -112,59 +170,63 @@ describe('Explore', () => {
...
 
         cy.findByText('Filters').parent().findByRole('button').click();
         cy.findByText('SQL').parent().findByRole('button').click();
+        cy.findByText('Charts').parent().findByRole('button').click(); // Close chart
 
         cy.findAllByText('Loading chart').should('have.length', 0);
         // Filter by raw
         cy.get('tbody > :nth-child(1) > :nth-child(2)').click();
-        cy.contains('Filter by "2020-08-11, 23:44:00:000 (+00:00)').click();
+        cy.contains('Filter by "2019-12-02, 00:00:00:000 (+00:00)').click();
```
Comment (rephus) [inline packages/e2e/cypress/e2e/dates.cy.ts:186]: it seems the dataset (or sorting) has changed

File: packages/e2e/cypress/e2e/dates.cy.ts [hunk 3]
```diff
@@ -197,7 +260,7 @@ describe('Explore', () => {
         cy.get('.bp4-code').contains(
             `(DATE_TRUNC('YEAR', "customers".created)) = ('2017-01-01')`,
         );
-        cy.get('[icon="chevron-up"]').click({ multiple: true });
+        cy.get('button[aria-label="increment"]').click({ multiple: true });
```
Comment (rephus) [inline packages/e2e/cypress/e2e/dates.cy.ts:263]: now we have multiple chevron-up on the page, so I need to be more specific here

File: packages/frontend/src/hooks/useQueryResults.tsx [hunk 4]
```diff
@@ -146,12 +146,14 @@ export const useUnderlyingDataResults = (
 // This hook will be used for getting charts in view mode and dashboard tiles
 export const useChartResults = (chartUuid: string, filters?: Filters) => {
     const queryKey = ['savedChartResults', chartUuid, JSON.stringify(filters)];
+    const timezoneFixQuery = convertDateFilters({ filters } as MetricQuery);
```
Comment (rephus) [inline packages/frontend/src/hooks/useQueryResults.tsx:149]: This is the fix !

**Review by owlas — APPROVED** [review]

**Pull request merged**